internal/server/server_test.go

```diff
@@ -45,6 +45,41 @@ func (o *oidcProviderRoundTripper) RoundTrip(r *http.Request) (*http.Response, e
 	return resp, nil
 }
 
+type healthzRoundTripper struct{}
```

Comment:
nit: The roundtrippers could be combined to a single one and logic added to switch the response based on the information supplied in the http.Request.